### DISCUSSION TITLE
Error show 'Enter a valid date' Can not filter with datetime type.

### DISCUSSION THREAD

**Discussion by technqvi** [carltongibson/django-filter#1393] [Q&A] [answered]
Hi,

**### I cannot use DateTime type  to filter(**_use widget=DateTimeInput_**)  .  the error was given that "Enter a valid date."**
it didn't work at all.
this is query using  MyDateTimeInput  :http://127.0.0.1:8000/error/?job_type=&start_datetime__gt=2021-02-14T01%3A00&end_datetime__lt=&error_type=
**_start_datetime__gt=2021-02-14T01%3A00_**

but  DateType I can do it (**_Use widget =DateInput_**).  it works
this is query using MyDateInput widget  : http://127.0.0.1:8000/error/?job_type=&start_datetime__gt=2021-02-14&end_datetime__lt=&error_type=
**_start_datetime__gt=2021-02-14_**

you can see in the figure below. 
I want to filter for listing item only where "Error From" is greater that 14 Feb 21 but   query result still return  12 Feb 21
**_I assume  2021-02-14T01%3A00 is the incorrect format as a parameter to query._**
**### _I don't know how to fix it, are there other solutions or  widgets for filtering by DateTime type._**
![image](https://user-images.githubusercontent.com/38780060/108377140-4cf72d80-7236-11eb-9513-2aec010505fc.png)

 
<img width="773" alt="sort_filter_page" src="https://user-images.githubusercontent.com/38780060/108361719-6cd22580-7225-11eb-8809-bf5f90b05150.png">

Belows are my source code.  **_you can see widget which I use  both DateInput  and  DateTimeInput  in start_datetime__gt
filter.py_**
```

import django_filters
from django_filters import *

from app.models import *
from django.forms import *

class ErrorLogFilter(django_filters.FilterSet):
    start_datetime__gt = django_filters.DateFilter(field_name='created_date', lookup_expr='gt',
                                                   widget=DateInput(format='%Y-%m-%d"',
                                                                        attrs={'type': 'date'}),
                                                   # widget=DateTimeInput(format='%d/%m/%Y %H:%M', attrs={'type': 'datetime-local'}),
                                                      label="Error From")
    end_datetime__lt = django_filters.DateFilter(field_name='created_date', lookup_expr='lt',
                                                             widget=DateInput(format='%Y-%m-%d"',
                                                                      attrs={'type': 'date'}),
                                                      label="End")
    error_type = filters.ModelChoiceFilter(queryset=ErrorLogType.objects.all(),
                                        field_name='type', label='Error Type')

    job_type=filters.ModelChoiceFilter(queryset=JobType.objects.all(),
                                        field_name='transaction__type', label='Job Type')


```

I try to use [https://pypi.org/project/django-datetime-widget2/](url) 
Configuration step is simple , then I try to apply ###  **_DateTimeWidget_** to  code filter.py  in filed start_datetime__gt 
```
     start_datetime__gt = django_filters.DateFilter(field_name='created_date', lookup_expr='gt',
                      
                                                   widgets=DateTimeWidget(attrs={'id':"ft_error_start_datetime_id",'type': 'datetime'},
                                                                          usel10n = True, bootstrap_version=3),

                                                      label="Error From")
```
error as below.
![image](https://user-images.githubusercontent.com/38780060/108398938-2512c480-724c-11eb-8d66-46b2ec838e6f.png)

**_So far, I can apply only  DateInput widget as code above, this is the only widget that  works for me_** 

     widget=DateInput(format='%Y-%m-%d"',
                                                                        attrs={'type': 'date'}),

![image](https://user-images.githubusercontent.com/38780060/108399230-791da900-724c-11eb-95ed-d323b555eb55.png)

thank you
Pongthorn

**Comment by Sunno** [comment]
Maybe the best solution is to use Django DateTime accepted formats https://www.geeksforgeeks.org/datetimefield-django-forms/
I tried to use a `datetime.isoformat()` generated datetime string and it wasn't working

**Comment by saadmk11** [comment, marked as answer]
You can use `DateTimeFilter` to filter using a DateTime string.

For Example:
```python
start_datetime__gt = django_filters.DateTimeFilter(field_name='created_date')
```

Ref: https://django-filter.readthedocs.io/en/main/ref/filters.html#datetimefilter

The error provided on the issue was caused by `widgets` because the filter takes `widget` as a keyword argument.
